social_core/backends/phabricator.py

```diff
@@ -0,0 +1,48 @@
+"""
+Phabricator OAuth2 backend, docs at:
+    https://secure.phabricator.com/book/phabcontrib/article/using_oauthserver/
+"""
+from .oauth import BaseOAuth2
+
+
+class PhabricatorOAuth2(BaseOAuth2):
+    """Phabricator OAuth authentication backend"""
+    name = 'phabricator'
+    API_URL = 'https://secure.phabricator.com'
+    AUTHORIZATION_URL = 'https://secure.phabricator.com/oauthserver/auth/'
+    ACCESS_TOKEN_URL = 'https://secure.phabricator.com/oauthserver/token/'
+    ACCESS_TOKEN_METHOD = 'POST'
+    REDIRECT_STATE = False
+    DEFAULT_SCOPE = []
```

Comment:
If there is no default scope, this could also be omitted, or better set to `None` same as the super class, but ... 

I suspect that the default scope is actually `scope.always`.  Check out  https://secure.phabricator.com/D15621 , which removed previous scopes `offline_access` and `whoami` as no longer needed, but added a new one.

Phabricator has no published defined scopes, as noted at https://github.com/ofbeaton/oauth2-phabricator#managing-scopes , and "This section has not been written yet." on https://secure.phabricator.com/book/phabcontrib/article/using_oauthserver/ .

If true, this deserves special mention as Phabricator is very odd in that regard; not many other OAuth providers have no scopes at all.

But as I mentioned above, it looks like the scope `scope.always` exists, and I wouldnt be surprised if the myriad other [scopes in phabricator](https://github.com/phacility/phabricator/search?utf8=%E2%9C%93&q=scope&type=) also work via OAuth.